Tasks/SqlAzureDacpacDeployment/Utility.ps1

```diff
@@ -18,36 +18,81 @@ function Get-AgentIPRange
     )
 
     [hashtable] $IPRange = @{}
-    $sqlCmdArgs = "-S $serverName -U $sqlUsername -P $sqlPassword -Q `"select getdate()`""
 
-    Write-Verbose "Reching SqlServer to check connection by running sqlcmd.exe $sqlCmdArgs"
+    if (Get-Command "SqlCmd.exe" -ErrorAction SilentlyContinue)
+    {
+        $sqlCmdArgs = "-S $serverName -U $sqlUsername -P $sqlPassword -Q `"select getdate()`""
+    
+        Write-Verbose "Reching SqlServer to check connection by running sqlcmd.exe $sqlCmdArgs"
 
-    $ErrorActionPreference = 'Continue' 
-    ( Invoke-Expression "&sqlcmd.exe --% $sqlCmdArgs" -ErrorVariable errors -OutVariable output 2>&1 ) | Out-Null
-    $ErrorActionPreference = 'Stop'
+        $ErrorActionPreference = 'Continue'
 
-    if($errors.Count -gt 0)
-    {
-        $errorMsg = $errors[0].Exception.Message
-        Write-Verbose $errorMsg
+        $sqlCmdPath = ( Get-Command -Name "SqlCmd.exe").Path
+        ( Invoke-Expression "& '$sqlCmdPath' --% $sqlCmdArgs" -ErrorVariable errors -OutVariable output 2>&1 ) | Out-Null   
 
-        $pattern = "([0-9]+).([0-9]+).([0-9]+)."
-        $regex = New-Object  -TypeName System.Text.RegularExpressions.Regex -ArgumentList $pattern
+        $ErrorActionPreference = 'Stop'
 
-        if($errorMsg.Contains("sp_set_firewall_rule") -eq $true -and $regex.IsMatch($errorMsg) -eq $true)
+        if($errors.Count -gt 0)
         {
-            $ipRangePrefix = $regex.Match($errorMsg).Groups[0].Value;
-        }
+            $errorMsg = $errors[0].Exception.Message
+            Write-Verbose $errorMsg
 
-        Write-Verbose "IP Range Prefix $ipRangePrefix"
+            $pattern = "([0-9]+).([0-9]+).([0-9]+)."
+            $regex = New-Object  -TypeName System.Text.RegularExpressions.Regex -ArgumentList $pattern
 
-        $IPRange.StartIPAddress = $ipRangePrefix + '0'
-        $IPRange.EndIPAddress = $ipRangePrefix + '255'
+            if($errorMsg.Contains("sp_set_firewall_rule") -eq $true -and $regex.IsMatch($errorMsg) -eq $true)
+            {
+                $ipRangePrefix = $regex.Match($errorMsg).Groups[0].Value;
+                Write-Verbose "IP Range Prefix $ipRangePrefix"
+
+                $IPRange.StartIPAddress = $ipRangePrefix + '0'
+                $IPRange.EndIPAddress = $ipRangePrefix + '255'
+            }
+        }
     }
+    else
+    {
+        Write-Verbose "SqlCmd.exe is not found in PATH" 
+
+        $IPRange.StartIPAddress = Get-AgentStartIPAddress
+        $IPRange.EndIPAddress = $IPRange.StartIPAddress
+    }    
 
     return $IPRange
 }
 
+
+function Get-AgentStartIPAddress
+{
+    $endpoint = (Get-VstsEndpoint -Name SystemVssConnection -Require)
+    $vssCredential = [string]$endpoint.auth.parameters.AccessToken
+
+    $vssUri = $env:SYSTEM_TEAMFOUNDATIONCOLLECTIONURI
```

Comment:
Yes this logic won't work for on prem. Next sprint I will add logic to detect sqlcmd.exe location and fix it. For now quickly unblock existing users I am doing this.